overlord/snapstate/snapstate_update_test.go

```diff
@@ -5164,28 +5190,18 @@ func (s *snapmgrTestSuite) TestUpdateManyWaitForBasesUC16(c *C) {
 		chg.AddAll(ts)
 	}
 
-	prereqTotal := len(tts[0].Tasks()) + len(tts[1].Tasks())
-	prereqs := map[string]bool{}
-	for i, task := range tts[2].Tasks() {
-		waitTasks := task.WaitTasks()
-		if i == 0 {
-			c.Check(len(waitTasks), Equals, prereqTotal)
-		} else if task.Kind() == "link-snap" {
-			c.Check(len(waitTasks), Equals, prereqTotal+1)
-			for _, pre := range waitTasks {
-				if pre.Kind() == "link-snap" {
-					snapsup, err := snapstate.TaskSnapSetup(pre)
-					c.Assert(err, IsNil)
-					prereqs[snapsup.InstanceName()] = true
-				}
-			}
-		}
-	}
-
-	c.Check(prereqs, DeepEquals, map[string]bool{
-		"core":      true,
-		"some-base": true,
-	})
+	// Some-snap is expected to wait for both the essential snap, but
+	// also the base of some-snap. These dependencies are set up between
+	// last tasks of core/some-base, on preprequisites
+	lastTaskOfCore, err := tts[0].Edge(snapstate.EndEdge)
+	c.Assert(err, IsNil)
+	lastTaskOfBase, err := tts[1].Edge(snapstate.EndEdge)
+	c.Assert(err, IsNil)
+	firstTaskOfSnap, err := tts[2].Edge(snapstate.BeginEdge)
+	c.Assert(err, IsNil)
+	c.Check(firstTaskOfSnap.WaitTasks(), HasLen, 2)
+	c.Check(firstTaskOfSnap.WaitTasks(), testutil.Contains, lastTaskOfCore)
+	c.Check(firstTaskOfSnap.WaitTasks(), testutil.Contains, lastTaskOfBase)
```

Comment:
should we check the expected lane situation here?

Reply:
added!